### PR TITLE
React UI

### PR DESCRIPTION
**Pull request by jimscarver**

### CODE REVIEW

**Comment by dckc** [comment]
This still deletes and re-creates files. Please stop using commit 564e09248c64b211ca9a73ac9ef1500ce2e8597f . Go back to master and commit from there.

Please review your diffs when you commit and write a commit message that corresponds to the diffs. If you're deleting a file, the commit message should be clear about why.

**Comment by jimscarver** [comment]
I cannot make all the changes tobi make from master. I did what I thought you told me to do.

**Comment by jimscarver** [comment]
the diffs do not show the files deleted  or restored as far as I can see in the diffs. Some files are deleted because I failed to replace them believing they were not in use. I re added Ballot.js so I guess I lost history on that. Perhaps that can be readded properly by you dan?

**Comment by dckc** [comment]
> the diffs do not show the files deleted or restored as far as I can see in the diffs.

I mean the diffs to each commit. The message for each commit should match the diffs for each commit.

> ... Perhaps that can be readded properly by you dan?

I went ahead and did what I suggested:
  - check out master in one directory, the react branch in another
  - copy files from the react director to the master directory
  - commit the results to a new branch
    - well, I re-used the branch name `reacttry` to go with this pull request.
    - the previous contents of reacttry, c8e336d, are available as reacttry-oops

I'm still not ready to merge; I'll comment separately.

File: package.json [hunk 1]
```diff
@@ -1,79 +1,39 @@
...
-      "import/prefer-default-export": "off"
-    }
-  },
-  "prettier": {
-    "trailingComma": "all",
-    "singleQuote": true
-  },
+  "name": "rholang-template",
+  "version": "1.0.0",
+  "description": "A template for building your dApp. Execute rholang code on the rchain network in real time.",
```
Comment (dckc) [inline package.json:4]: Please fix the description.

File: package.json [hunk 2]
```diff
@@ -1,79 +1,39 @@
...
-    "eslint-plugin-import": "^2.22.0",
-    "eslint-plugin-prettier": "^3.1.4",
-    "html-webpack-plugin": "^4.5.0",
-    "prettier": "^2.1.1",
-    "typescript": "^4.0.2",
-    "webpack": "^4.44.2",
-    "webpack-cli": "^3.3.12",
-    "webpack-dev-server": "^3.11.0"
-  },
-  "eslint": {
```
Comment (dckc) [inline package.json:32]: Why delete the eslint rules? I should have documented the project style in CONTRIBUTING.md as well, but I'd like this project to match agoric / airbnb style.

File: package.json [hunk 3]
```diff
@@ -1,79 +1,39 @@
...
-  "scripts": {
-    "lint-fix": "eslint --fix '**/*.{js,jsx}'",
-    "lint-check": "eslint '**/*.{js,jsx}'",
-    "lint:types": "tsc -p jsconfig.json",
-    "start": "webpack-dev-server --open",
-    "build": "webpack"
-  },
-  "contributors": [
-    "Jim Whitescarver",
-    "Dan Connolly"
```
Comment (dckc) [inline package.json:15]: Taking our names out of the contributors list is extremely un-cool.

File: package.json [hunk 4]
```diff
@@ -1,79 +1,39 @@
 {
-  "name": "liquid-democracy-rchain",
-  "version": "0.0.1",
-  "description": "",
-  "main": "Jakefile.js",
-  "scripts": {
-    "lint-fix": "eslint --fix '**/*.{js,jsx}'",
-    "lint-check": "eslint '**/*.{js,jsx}'",
-    "lint:types": "tsc -p jsconfig.json",
```
Comment (dckc) [inline package.json:9]: The lint:types rule is important, along with lint-fix and lint-check. Why were they removed?

File: public/index.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
+</body>
+
+</html>
```
Comment (dckc) [inline public/index.html:43]: newline missing at end of file

**Review by dckc — CHANGES_REQUESTED** [review]
Without instruction on how to run it, I can't evaluate this further.

I tried `yarn install; yarn start` but I got:

```
SyntaxError: /tmp/ld-master/jsconfig.json: Unexpected token / in JSON at position 0
```

**Comment by dckc** [comment]
ok... I got it running, after removing comments and trailing commas from `jsconfig.json`. I hope we can get whatever is tripping over those to ignore that file; it's annoying to lose comments. But I guess they're not criticial.

The README should say how to start it, though `yarn; yarn start` is sufficiently widely known that it's not critical at this stage.

**Comment by dckc** [comment]
@9rb the snyk thing is making noise again. The Details link goes to something that I don't have access to.

As I [previously asked][1], please turn it off.

[1]: https://github.com/rchain-community/rchain-api/pull/83#issuecomment-786342584

**Comment by dckc** [comment]
Is the UI interestingly different with react? The main difference I saw was a bunch more text that made the main widgets further down the page.

If we're going to switch UI libraries, perhaps this PR should add something that show off the difference?

I'm not sure I want to take on the risk of **30x dependencies**. It's _much_ slower to start up.

excluding devDependencies (`npm install --only=prod`) comparing bloat vs master...

|             | deps | size |
|-----------| ------ | ------ |
| master  | 118  | 44M |
| react  |  3429 | 281M |

```
$ du -sh node_modules/
281M	node_modules/

$ npm ls | wc -l
3429
```



```
$ du -sh node_modules/
44M	node_modules/

$ npm ls | wc -l
118
```

**Comment by jimscarver** [comment]
the react overhead is quite high and without much benefit..

**Pull request closed without merging**